File: files/400-499/452/Challenge 452.py
import numpy as np
import pandas as pd

p = "400-499/452/CH-452 Matrix Normalization.xlsx"
matrix = pd.read_excel(p, sheet_name="Sheet1", header=None).iloc[3:7, 2:6].to_numpy()
expected = pd.read_excel(p, sheet_name="Sheet1", header=None).iloc[3:7, 9:13].to_numpy()

result = np.sort(np.sort(matrix, axis=1), axis=0)
print((result == expected).all())

File: files/400-499/452/Challenge 452.R
library(tidyverse)
library(readxl)

p <- "400-499/452/CH-452 Matrix Normalization.xlsx"
matrix <- read_excel(
  p,
  sheet = "Sheet1",
  range = "C4:F7",
  col_names = FALSE
) %>%
  as.matrix()
expected <- read_excel(
  p,
  sheet = "Sheet1",
  range = "J4:M7",
  col_names = FALSE
) %>%
  as.matrix()

result <- matrix %>%
  apply(1, sort) %>%
  t() %>%
  apply(2, sort)

print(isTRUE(all.equal(unname(result), unname(expected))))
